syncopy/tests/test_specest_fooof.py

```diff
@@ -0,0 +1,184 @@
+                                                           fadeIn=fadeIn, fadeOut=fadeOut, short=True)
+    cfg = get_defaults(freqanalysis)
+    cfg.method = "mtmfft"
+    cfg.taper = "hann"
+    cfg.select = {"trials": 0, "channel": 1}
+    cfg.output = "fooof"
+
+    def test_fooof_output_fooof_fails_with_freq_zero(self, fulltests):
+        """ The fooof package ignores input values of zero frequency, and shortens the output array
+            in that case with a warning. This is not acceptable for us, as the expected output dimension
+            will not off by one. Also it is questionable whether users would want that. We therefore use
+            consider it an error to pass an input frequency axis that contains the zero, and throw an
+            error in the frontend to stop before any expensive computations happen. This test checks for
+            that error.
+        """
+        self.cfg['output'] = "fooof"
+        self.cfg['foilim'] = [0., 250.]    # Include the zero in tfData.
+        self.cfg['out'] = None
+        with pytest.raises(SPYValueError) as err:
+            _ = freqanalysis(self.cfg, self.tfData)  # tfData contains zero.
+        assert "a frequency range that does not include zero" in str(err.value)
+
+    def test_fooof_output_fooof_works_with_freq_zero_in_data_after_setting_foilim(self, fulltests):
+        """
+        This tests the intended operation with output type 'fooof': with an input that does not
+        include zero, ensured by using the 'foilim' argument/setting when calling freqanalysis.
+
+        This returns the full, fooofed spectrum.
+        """
+        self.cfg['output'] = "fooof"
+        self.cfg['foilim'] = [0.5, 250.]    # Exclude the zero in tfData.
+        self.cfg['out'] = None
+        spec_dt = freqanalysis(self.cfg, self.tfData)
+
+        # check frequency axis
+        assert spec_dt.freq.size == 500
+        assert spec_dt.freq[0] == 0.5
+        assert spec_dt.freq[499] == 250.
+
+        # check the log
+        assert "fooof_method = fooof" in spec_dt._log
+        assert "fooof_aperiodic" not in spec_dt._log
+        assert "fooof_peaks" not in spec_dt._log
+        assert "fooof_opt" in spec_dt._log
+
+        # check the data
+        assert spec_dt.data.ndim == 4
+        assert spec_dt.data.shape == (1, 1, 500, 1)
+        assert not np.isnan(spec_dt.data).any()
+
+        # check that the cfg is correct (required for replay)
+        assert spec_dt.cfg['freqanalysis']['output'] == 'fooof'
+
+        # Plot it.
+        #  _plot_powerspec(freqs=spec_dt.freq, powers=spec_dt.data[0, 0, :, 0])
+        #spec_dt.singlepanelplot()
+
+    def test_spfooof_output_fooof_aperiodic(self, fulltests):
+        """Test fooof with output type 'fooof_aperiodic'. A spectrum containing only the aperiodic part is returned."""
+        self.cfg['output'] = "fooof_aperiodic"
+        self.cfg['foilim'] == [0.5, 250.]
```

Comment:
This can't work right, you want an assignment here!? Wondering how this tests actually passes, and indeed if I manually do:
```python
T = TestFooofSpy()
T.test_spfooof_output_fooof_aperiodic()
```
this fails with the 0-frequency error, how does this pass with pytest??

Reply:
I think it works because of the cfg bug, which is still present in that branch. the cfg has been changed globally by another test, and it stays for the next one.

Reply:
I will merge the current `dev` into the branch at the very end. then, stuff like this will get noticed.